woocommerce/woocommerce-gutenberg-products-block
# Update CSS build process

Fixes #396 – we want to have a distinction between frontend (general) and backend (editor) styles, so now we can create `style.scss` and `editor.scss` for each block. The CSS in `style.scss` will be loaded in both the editor and the frontend (so styled blocks are previewed correctly). The CSS in `editor.scss` will only be loaded in the editor, so it's used for the placeholders, edit mode, etc. This split will let us style blocks without accidentally bringing over editor styles into the frontend (this is what caused #335 ).

We're now building all the CSS down into two files, `style.css` and `editor.css`. These are shared across all blocks.

The last big change is a new custom webpack plugin to handle a side effect of `MiniCssExtractPlugin`– this also generates a small required JS file per chunk, so we take the content and merge it into the `vendors.js` file. If we don't have this code, the blocks will silently fail 🙃 

### How to test the changes in this Pull Request:

1. Run `npm run build` or `npm start`
2. Expect: The `build` directory should contain only two CSS files
3. Expect: There shouldn't be a "style.js" or "editor.js" file
4. Load up the editor, try adding some of our blocks
5. Expect: the blocks are available, and there are no style changes
6. View a post with the Featured Product block on the front end of the site
7. Expect: It should appear as normal, no style changes
